File: AdventOfCodeC#/Day1.cs
namespace Day1;

public static class Day1
{
    public static int Execute(string[] input) => input.Aggregate((0,0), (x, y) => Process(y, x)).Item1;

    private static (int highestTotal, int runningTotal) Process(string input, (int highestTotal, int runningTotal) totals) =>
        input == "" 
            ? (Math.Max(totals.highestTotal, totals.runningTotal), 0) 
            : (totals.highestTotal, (totals.runningTotal + int.Parse(input)));
}

File: AdventOfCode/Day1.fs
module Day1

open System;

let f2 ((highest:int, total:int), item:string) =  
    if item = "" 
    then 
        (Math.Max(highest,total), 0)
    else            
        (highest, (total + (item |> int)))

let execute (input:string[]) =
    let (a, _) = Array.fold(fun item result -> f2(item, result)) (0,0) input
    printfn $"{a}"